File: typescript-connector/src/models/Properties.ts
// Class from DFC Business Ontology: #Properties
import { SemanticObject } from "../core/SemanticObject.js";

export interface PropertiesParams {
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
}
export class Properties extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Properties";
  }

  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;

  constructor(
    semanticId: string,
    params?: PropertiesParams,
  ) {
    super(semanticId);
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.semanticType = Properties.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:Properties:date", () => this.date);
    this.registerSemanticProperty("dfc-b:Properties:description", () => this.description);
    this.registerSemanticProperty("dfc-b:Properties:name", () => this.name);
    this.registerSemanticProperty("dfc-b:Properties:characteristic_of", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:Properties:has_dimension", () => this.hasDimension);
  }
  static {
    SemanticObject.typeRegistry.set(Properties.SEMANTIC_TYPE, Properties);
  }
}

File: ruby-gem/lib/models/properties.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Properties
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class Properties < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:Properties".freeze

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:Properties"
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
